docker-compose.yaml

```diff
@@ -1,59 +1,64 @@
 version: '3'
-
-# Other services are specified in .\lib\killrvideo-docker-common\docker-compose.yaml
 services:
-  node:
-    build: /Users/aleksandrvolochnev/killrvideo/node
+  # NodeJS Backend
+  backend:
+    build: .
+    volumes:
+      - .:/usr/src/app
     depends_on:
       - dse
     environment:
-      KILLRVIDEO_ETCD: "etcd:2379"
-      KILLRVIDEO_DSE_USERNAME: $KILLRVIDEO_DSE_USERNAME
-      KILLRVIDEO_DSE_PASSWORD: $KILLRVIDEO_DSE_PASSWORD
-      KILLRVIDEO_CASSANDRA_REPLICATION: $KILLRVIDEO_CASSANDRA_REPLICATION
-      KILLRVIDEO_LOGGING_LEVEL: $KILLRVIDEO_LOGGING_LEVEL
+      KILLRVIDEO_LOGGING_LEVEL: debug
 
-  # Start the KillrVideo web UI on port 3000
+  # Datastax Enterprise 
+  dse:
+    image: datastax/dse-server:6.0.0
+    command: [ -s -g ]
+    ports:
+      - "9042:9042"
+      - "8983:8983"
+      - "8182:8182"
+    environment:
+      DS_LICENSE: accept
+    # Allow DSE to lock memory with mlock
+    cap_add:
+      - IPC_LOCK
+    ulimits:
+      memlock: -1
+
+  # Frontend
   web:
-    image: killrvideo/killrvideo-web:1.2.6
-    #build: /Users/aleksandrvolochnev/killrvideo/web
+    image: killrvideo/killrvideo-web:3.0.0-rc1
     ports:
     - "3000:3000"
     depends_on:
     - dse
-    - etcd
     environment:
-      SERVICE_3000_NAME: web
-      KILLRVIDEO_ETCD: "etcd:2379"
-      KILLRVIDEO_DSE_USERNAME: $KILLRVIDEO_DSE_USERNAME
-      KILLRVIDEO_DSE_PASSWORD: $KILLRVIDEO_DSE_PASSWORD
-      KILLRVIDEO_CASSANDRA_REPLICATION: $KILLRVIDEO_CASSANDRA_REPLICATION
-      KILLRVIDEO_LOGGING_LEVEL: $KILLRVIDEO_LOGGING_LEVEL
+      KILLRVIDEO_LOGGING_LEVEL: debug
```

Comment:
Same comment as above regarding using dynamic environment variables.

Reply:
Right, I don't do this explicitly in docker-compose because these values are all dynamic depending on the environment we are working with, even for local developer builds, not just the cloud deployment.  I also will not store username or password information in docker-compose or any other config files that are committed into any repo.  If we take username and password there are multiple use cases, one using the cassandra defaults, one with a different application username, another with an admin superuser different from the default cassandra user, and then our external cloud deployments and any time an external cluster is used.  In each of those cases are different combinations of username and password.  I don't think it makes sense to maintain different docker-compose.yamls for each and I wouldn't store than info in them anyway.  Simply enough, if anything other than the default cassandra creds are used for DSE then the backend, generator, and webUI must all know about these changes if authentication is enabled or they will fail to connect to DSE.  By setting environ vars instead of hardcoding these into config files I can ensure they are propagated everywhere they need to be without modifying any files.

Reply:
I do agree that using environment variables is not as obvious as setting directly in the yamls.  Notice though I do not do this for every configuration option, only those that have a need to 1) be shared across multiple services, 2) and/or are used in the cloud deployment, 3) have a need to be dynamic.